### ISSUE TITLE
Collate function confusion

### ISSUE THREAD

**Issue by KayARS** [webdataset/webdataset#210]
Hello,
im trying to figure out how the [default_collation_fn](https://webdataset.github.io/webdataset/api/webdataset/iterators.html) works:
The first line of comment says
   ` """Take a collection of samples (dictionaries) and create a batch.`
But the first line of code checks  that the sample is a list or tuple, not a dictionary
  `  assert isinstance(samples[0], (list, tuple)), type(samples[0])
`
With the first comment I would expect _samples_ to be a list of dictionaries, but it should be a list
Im loading about 20 numpy arrays which are in one dictionary of the form
{"data1.npy": [np array data],...}

Do I have to use "to_tuple"? It would be quite a pain to name each sample individually in this function
Am I missing something here? 

Thanks a lot in advance

**Comment by tmbdev** [comment]
Yes, sorry, the current collation function doesn't handle dictionaries.

I'll add that as an enhancement.

**Comment by jenuk** [comment]
Starting in pytorch 1.11 there is [`torch.utils.data.default_collate`](https://pytorch.org/docs/stable/data.html#torch.utils.data.default_collate). That might be a solution for your problem @KayARS (if you still have this problem) since you can give a collate function to batched.

Related to that is the `unbatch` function [here](https://github.com/webdataset/webdataset/blob/cb070965b9919f9e9e942a7c87e8227284383e5f/webdataset/filters.py#L511) that also does not handle batches in the dict format. Would you be interested in a pull request that enables dicts there?

**Comment by tmbdev** [comment]
I'll have a look at a PR.

Note that you can pass custom collate functions, so it's just a one liner anyway.
